docs/syntax-jsx-support.md

```diff
@@ -0,0 +1,338 @@
+local State = Fusion.Value("Hello")
+
+local ui =
+    <textlabel
+        Text={State}
+        Size={UDim2.fromScale(1, 1)}
+    />
+```
+
+This example illustrates the key point: **JSX is a “call tree” surface syntax**, and Fusion (not Luau) defines the semantics.
+
+### Example: non-UI usage (scene graph / “fiber-like” tree)
+
+JSX can also be used for non-UI declarative graphs, e.g. building a 3D scene tree out of Roblox Instances (a “fiber-like” idea: a declarative tree that produces runtime objects).
+
+```lua
+local Intrinsics = {
+    model = "Model",
+    part = "Part",
+    pointlight = "PointLight",
+    attachment = "Attachment",
+    -- ...
+}
+
+-- A minimal Instance-producing factory:
+function jsx(tag, props, ...)
+    assert(type(tag) == "string", "This example expects intrinsic (lowercase) tags")
+    local className = assert(Intrinsics[tag], ("Unknown intrinsic tag %q"):format(tag))
+
+    local inst = Instance.new(className)
+    props = props or {}
+
+    -- Apply properties
+    for k, v in pairs(props) do
+        inst[k] = v
+    end
+
+    -- Parent children to this instance
+    for _, child in ipairs({ ... }) do
+        if typeof(child) == "Instance" then
+            child.Parent = inst
+        end
+    end
+
+    return inst
+end
+
+local scene =
+    <model Name="Root">
+        <part Name="Ground" Anchored={true} Size={Vector3.new(50, 1, 50)} />
+        <part Name="Lamp" Position={Vector3.new(0, 5, 0)}>
+            <pointlight Brightness={3.5} Range={24} />
+        </part>
+    </model>
+```
+
+This keeps the same core idea: the tag is a string, props are a table, and children are a list. The meaning is defined by the chosen runtime factory (`jsx`), and JSX is simply a concise way to express the tree.
+
+### Compatibility & ambiguity analysis
+
+#### Backwards compatibility
+
+With `.luax` + flag gating, existing `.lua`/`.luau` source is unaffected.
+
+Within `.luax`, code that uses `<` for comparisons (e.g. `a < b`) remains valid; JSX is only recognized when the parser expects a `simpleexp` and sees `<` as the next token in that position.
+
+#### Grammar ambiguity concerns
+
+Key ambiguity sources:
+
+- `<` as binary operator vs JSX element start
+- `</` sequences inside other contexts
+- Interaction with existing type syntax, including explicit type instantiation (`<<T>>()`)
+
+Mitigations:
+
+- Restrict JSX recognition to positions where a `simpleexp` is parsed.
+- Keep JSX syntax small initially (no raw text children, fragments, or spreads).
+- Continue gating behind a flag to allow collecting real-world compatibility data.
+
+### Editor integration & tooling considerations
+
+Adding JSX affects:
+
+- Tokenization/highlighting for `.luax`
+- Parsing for autocomplete, formatter/pretty-printer, and incremental parse
+- Error recovery expectations (JSX introduces new “paired delimiter” structures)
+
+Mitigations:
+
+- Keep JSX AST lowering to existing nodes so downstream tools operate on familiar AST shapes.
```

Comment:
You earlier argued that this would not use a compilation step, and then this sentence is literally describing a compilation step. This clouds my understanding of what you intend a lot here.

Reply:
The point being made here is that there are no new AST nodes to deal with here either internally by the interpreter or externally if there are other tools that do depend on luau's AST. I'm not arguing for a build step as part of this proposal.

That said, as mentioned in another comment, it may be *desirable* to introduce new AST nodes for exactly this reason - so external consumers can leverage them for things like syntax highlighting, formatting, etc. this should be revised to state there 

should revised to say no new *CST nodes*, but would likely introduce new AST ones

(I'm making some assumptions about the internal workings of luau I need to double check)